components/ProductSelector/index.tsx

```diff
@@ -0,0 +1,454 @@
+		event: React.ChangeEvent<HTMLInputElement>
+	) => {
+		setKeyword(event.target.value)
+	}
+
+	const selectProduct = (product: any) => {
+		if (
+			ampSDK?.type === 'array' &&
+			ampSDK.maxItems &&
+			selectedProducts.length >= ampSDK.maxItems
+		) {
+			setAlertMessage(
+				"You've reached the maximum amount of selectable items"
+			)
+			setShowAlert(true)
+		} else {
+			// TODO: [NOVADEV-980] Update UX to validate if user wants to add variants of a master
+			const match = selectedProducts.find(
+				(p) => p.selectedVariant.sku === product.variants[0].sku
+			)
+			//console.log('match: ', match)
+			if (match) {
+				setAlertMessage("You've already selected this item")
+				setShowAlert(true)
+			} else if (ampSDK?.type === 'string' || ampSDK?.type === 'object') {
+				// For single selection, replace the selected product.
+				setSelectedProducts((selectedProducts) => [
+					{
+						...product,
+						deleteKey: selectedProducts.length,
+						selectedVariant: product.variants[0],
+					},
+				])
+			} else {
+				setSelectedProducts((selectedProducts) => [
+					...selectedProducts,
+					{
+						...product,
+						deleteKey: selectedProducts.length,
+						selectedVariant: product.variants[0],
+					},
+				])
+			}
+		}
+	}
+
+	const removeProduct = (productTile: any) => {
+		setSelectedProducts(
+			selectedProducts.filter(
+				(p) => p.deleteKey !== productTile.deleteKey
+			)
+		)
+	}
+
+	//const updateVariant()
+
+	const getContainerHeight = () => {
+		const newHeight = container.current.clientHeight
+		setHeight(newHeight)
+	}
+
+	const updateSelected = useCallback(
+		(selectedProducts) => {
+			if (selectedProducts.length) {
+				switch (ampSDK?.type) {
+					case 'string':
+						const formStr = selectedProducts.map((prod) => prod.id)
+						ampSDK.setValue(formStr[0])
+						break
+					case 'strings':
+						const formStrs = selectedProducts.map((prod) => prod.id)
+						ampSDK.setValue(formStrs)
+						break
+					case 'object':
+						const formVal = {
+							id: selectedProducts[0].id,
+							variant: selectedProducts[0].selectedVariant?.sku,
+						}
+						ampSDK.setValue(formVal)
+						break
+					case 'objects':
+						const formVals = selectedProducts.map((prod) => ({
+							id: prod.id,
+							variant: prod.selectedVariant?.sku,
+						}))
+						ampSDK.setValue(formVals)
+						break
+
+					default:
+						break
+				}
+			} else {
+				ampSDK.clearValue()
+			}
+		},
+		[ampSDK]
+	)
+
+	useEffect(() => {
+		if (!mode) keywordInput.current.value = ''
+		setResults([])
+	}, [mode])
+
+	useEffect(() => {
+		setPage(1)
+		setNoOfPages(Math.ceil(results.length / itemsPerPage))
+		setTimeout(() => {
+			getContainerHeight()
+		}, 100)
+		//console.log('results: ', results)
+	}, [results, selectedProducts])
+
+	useEffect(() => {
+		window.addEventListener('resize', getContainerHeight)
+
+		return () => {
+			window.removeEventListener('resize', getContainerHeight)
+		}
+	}, [])
```

Comment:
getContainerHeight is not captured here. Maybe that should also be a useCallback dependant on `container`.

Reply:
I think I tried having that useEffect depend on container and it wasn't working..